pkg/client/pre.go

```diff
@@ -6,6 +6,77 @@ import (
 	p4_v1 "github.com/p4lang/p4runtime/go/p4/v1"
 )
 
+type CloneSessionOptions struct {
+	// ClassOfService defines class_of_service that should be set for cloned packets.
+	ClassOfService uint32
+	// PacketLenBytes defines packet size (in bytes) for cloned packets.
+	// Should be set to non-zero value if packets should be truncated.
+	PacketLenBytes int32
+}
+
+var DefaultCloneSessionOptions = CloneSessionOptions{
+	ClassOfService: 0,
+	PacketLenBytes: 0,
+}
+
+func (c *Client) InsertCloneSession(ctx context.Context, id uint32, ports []uint32, options CloneSessionOptions) error {
+	entry := &p4_v1.CloneSessionEntry{
+		SessionId:         id,
+		ClassOfService:    options.ClassOfService,
+		PacketLengthBytes: options.PacketLenBytes,
+	}
+
+	for _, port := range ports {
+		replica := &p4_v1.Replica{
+			EgressPort: port,
+			Instance:   uint32(0),
```

Comment:
there is a discrepancy between the `Instance` value for cloning and the `Instance` value for multicast, and I wonder if it's intentional or not? Ideally this should be exposed to the library client eventually, but in the mean time, I feel like the multicast way (using sequential `Instance` values) has one advantage as it will allow duplicates in the ports list.

Reply:
Yeah, you're right. I wasn't sure about this part, fixing it now.